### ISSUE TITLE
test: PgMiniTest.SystemTableTxnTest takes too long

### ISSUE THREAD

**Issue by jaki** [yugabyte/yugabyte-db#7657]
C++ test `PgMiniTest.SystemTableTxnTest` takes way too long.  On centos-gcc-debug, I got

```sh
./yb_build.sh --cxx-test pgwrapper_pg_mini-test --gtest_filter PgMiniTest.SystemTableTxnTest
```

```
[       OK ] PgMiniTest.SystemTableTxnTest (957676 ms)
```

which is about 16 minutes.  It has

```cpp
  for (int i = 1; i <= 100; ++i) {
```

and I think this can be reduced down to 50 at least, theoretically cutting time in half.  That's still 8 minutes, though, which is bad, so try to cut it down more without reducing test stability.

Later, it does

```
  ASSERT_GE(commit1_fail_count, 25);
  ASSERT_GE(insert2_fail_count, 25);
```

These should be using variables so that they are relative to the total iterations (namely, 1/4 times the number of iterations).

Use `--test-parallelism 1 --num-repetitions 1000` to check stability.  Test parallelism 1 is usually needed because it gets finicky otherwise.

keywords: PgMiniTest, SystemTableTxnTest

**Comment by jackstenglein** [comment]
Hello, I'd like to work on this issue, if it's still available?

**Comment by jaki** [comment]
Yeah, it is.  Also make sure to check stability on build types like asan, tsan.  Thanks.

**Comment by jpavani7** [comment]
Hello, I'd like to help. How can I get a copy of the files? It's my first time on github, but I'm excited and I want to help however I can.
